pennylane_lightning/core/_serialize.py

```diff
@@ -95,29 +100,52 @@ def __init__(
         else:
             raise DeviceError(f'The device name "{device_name}" is not a valid option.')
 
-        if device_name == "lightning.tensor":
-            self.tensornetwork_c64 = lightning_ops.TensorNetC64
-            self.tensornetwork_c128 = lightning_ops.TensorNetC128
-        else:
+        self.tensor_backend = tensor_backend
+        self._use_mpi = use_mpi
+
+        if device_name in ["lightning.qubit", "lightning.kokkos", "lightning.gpu"]:
             self.statevector_c64 = lightning_ops.StateVectorC64
             self.statevector_c128 = lightning_ops.StateVectorC128
 
-        self.named_obs_c64 = lightning_ops.observables.NamedObsC64
-        self.named_obs_c128 = lightning_ops.observables.NamedObsC128
-        self.hermitian_obs_c64 = lightning_ops.observables.HermitianObsC64
-        self.hermitian_obs_c128 = lightning_ops.observables.HermitianObsC128
-        self.tensor_prod_obs_c64 = lightning_ops.observables.TensorProdObsC64
-        self.tensor_prod_obs_c128 = lightning_ops.observables.TensorProdObsC128
-        self.hamiltonian_c64 = lightning_ops.observables.HamiltonianC64
-        self.hamiltonian_c128 = lightning_ops.observables.HamiltonianC128
+            self.named_obs_c64 = lightning_ops.observables.NamedObsC64
+            self.named_obs_c128 = lightning_ops.observables.NamedObsC128
+            self.hermitian_obs_c64 = lightning_ops.observables.HermitianObsC64
+            self.hermitian_obs_c128 = lightning_ops.observables.HermitianObsC128
+            self.tensor_prod_obs_c64 = lightning_ops.observables.TensorProdObsC64
+            self.tensor_prod_obs_c128 = lightning_ops.observables.TensorProdObsC128
+            self.hamiltonian_c64 = lightning_ops.observables.HamiltonianC64
+            self.hamiltonian_c128 = lightning_ops.observables.HamiltonianC128
 
-        if device_name != "lightning.tensor":
             self.sparse_hamiltonian_c64 = lightning_ops.observables.SparseHamiltonianC64
             self.sparse_hamiltonian_c128 = lightning_ops.observables.SparseHamiltonianC128
 
-        self._use_mpi = use_mpi
-
-        if self._use_mpi:
+        elif device_name == "lightning.tensor" and self.tensor_backend == "mps":
```

Comment:
Could we rewrite this as follows using two helper protected methods:
```
if device_name == LT:
    _set_lightning_tensor_bindings(backend)
else:
    _set_lightning_state_bindings()
```

Reply:
Done!